### PR TITLE
Add NXlog collectors and add new documentation reference

### PR DESCRIPTION
**Pull request by mpfz0r**
Add NXlog default collectors and templates for Linux and windows

Replace inline filebeat documentation with external links
We don't want to maintain the documentation of various collectors.
Link to the official documentation instead.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 1]
```diff
@@ -104,13 +104,154 @@ public void upgrade() {
                         "   - name: Security"
         );
         ensureCollector(
-                "nxlog",
+                "nxlog-linux",
```
Comment (mariussturm) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: Unfortunately this resolves UI-wise to `nxlog-linux on Linux` when you select the collector on the configuration page.

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 2]
```diff
@@ -104,13 +104,154 @@ public void upgrade() {
...
+                        "<Extension syslogExt>\n" +
+                        "  Module xm_syslog\n" +
+                        "</Extension>\n" +
+                        "\n" +
+                        "User nxlog\n" +
+                        "Group nxlog\n" +
+                        "\n" +
+                        "Moduledir /usr/lib/nxlog/modules\n" +
+                        "CacheDir /var/spool/collector-sidecar/nxlog\n" +
+                        "PidFile /var/run/graylog-sidecar/nxlog.pid\n" +
```
Comment (mariussturm) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: Do we want to use Sidecar specific paths for collectors or maybe rely completely on default paths?

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 3]
```diff
@@ -104,13 +104,154 @@ public void upgrade() {
...
+                        "\tReadFromLast True\n" +
+                        "\tRecursive False\n" +
+                        "\tRenameCheck False\n" +
+                        "\tExec $FileName = file_name(); # Send file name with each message\n" +
+                        "</Input>\n" +
+                        "\n" +
+                        "<Input syslog-udp>\n" +
+                        "\tModule im_udp\n" +
+                        "\tHost 127.0.0.1\n" +
+                        "\tPort 514\n" +
```
Comment (mariussturm) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: I wouldn't start a syslog input by default. We should either use an unprivileged port for this or comment the input if we really want to show how this works.

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 4]
```diff
@@ -104,13 +104,154 @@ public void upgrade() {
...
+                        "  Path file => gelf\n" +
+                        "</Route>\n" +
+                        "<Route route-2>\n" +
+                        "  Path syslog-udp => gelf\n" +
+                        "</Route>\n" +
+                        "\n" +
+                        "\n"
+        );
+        ensureCollector(
+                "nxlog-windows",
```
Comment (mariussturm) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: same as above

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java (hunk 2 above)
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: I took these from the config generated by the old sidecar.
Never change a running system ;-)

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java (hunk 3 above)
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: Ack. I'll comment this out

**Pull request merged**